tests/system.py

```diff
@@ -1126,6 +1126,62 @@ def test_load_table_from_dataframe_w_struct_datatype(self):
         self.assertEqual(table.schema, table_schema)
         self.assertEqual(table.num_rows, 3)
 
+    @unittest.skipIf(
+        pyarrow is None or PYARROW_INSTALLED_VERSION < PYARROW_MINIMUM_VERSION,
+        "Only `pyarrow version >=2.0.0` is supported",
+    )
+    @unittest.skipIf(pandas is None, "Requires `pandas`")
+    def test_load_table_from_dataframe_w_array_datatype(self):
+        """Test that a DataFrame contains array can be uploaded if a
+        BigQuery without specifying a schema.
+
+        https://github.com/googleapis/python-bigquery/issues/19
+        """
+        table_schema = [
+            bigquery.SchemaField(
+                "A",
+                "RECORD",
+                "NULLABLE",
+                None,
+                (
+                    bigquery.SchemaField(
+                        "list",
+                        "RECORD",
+                        "REPEATED",
+                        None,
+                        (
+                            bigquery.SchemaField(
+                                "item", "INTEGER", "NULLABLE", None, (), None
```

Comment:
Hmm... This is a bit of a surprising schema. It appears to match the behavior we were encountering previously. This feature is not supported if we cannot upload directly to a REPEATED INTEGER column.